components/Button/src/index.tsx

```diff
@@ -44,17 +33,22 @@ export interface ButtonProps extends BaseProps {
 /**
  * Primary UI component for user interaction
  */
-export const Button: React.FC<ButtonProps> = ({ iconAlign = 'start', ...props }: ButtonProps) => {
-  const rootClassNames = clsx('denhaag-button', {
-    'denhaag-button--secondary-action': props.variant === 'secondary-action',
-    'denhaag-button--large': props.size === 'large',
-    'denhaag-button--start-icon': iconAlign === 'start' && props.icon !== undefined,
-    'denhaag-button--end-icon': iconAlign === 'end',
-  });
+export const Button: React.FC<ButtonProps> = ({ type = 'button', iconAlign = 'start', ...props }: ButtonProps) => {
+  const className = clsx(
+    'denhaag-button',
+    {
+      'denhaag-button--secondary-action': props.variant === 'secondary-action',
+      'denhaag-button--large': props.size === 'large',
+      'denhaag-button--start-icon': iconAlign === 'start' && props.icon !== undefined,
+      'denhaag-button--end-icon': iconAlign === 'end',
+    },
+    props.className,
+  );
+
   const iconWrapped = props.icon !== undefined ? <span className={'denhaag-button__icon'}>{props.icon}</span> : '';
 
   return (
-    <button className={rootClassNames} onClick={props.onClick} disabled={props.disabled} type={props.type}>
+    <button {...props} type={type} className={className} onClick={props.onClick}>
```

Comment:
Is it necessary to have onClick written out here (as it is also covered by the spreaded `props`)?

Reply:
Also feel a bit weirdly about the `className` being set by the `props` and then overwritten by specifying it to the `className` const, but guess it works

Reply:
Fixed